### PR TITLE
os/arch/src/amebasmart: Add multiple SPI support

### PR DESCRIPTION
**Pull request by kaikaikaime**
Add menuconfig selection for multiple SPI
Change pins and roles

### CODE REVIEW

**Comment by sunghan-chang** [comment]
@kaikaikaime As far as I know, your team was working on new SDK porting. Is it ok to merge even new SDK will change this?

**Comment by kaikaikaime** [comment]
@sunghan-chang Yes, as this is the modification for the porting layer and not the new SDK library there should not be any conflict.

File: os/arch/arm/src/amebasmart/Kconfig [hunk 1]
```diff
@@ -71,11 +71,16 @@ config AMEBASMART_SPI
 	default n
 	
 if AMEBASMART_SPI
+config RTL8730E_SPI0
+	bool "Enable SPI0"
+	default n
+config RTL8730E_SPI1
```
Comment (sunghan-chang) [inline os/arch/arm/src/amebasmart/Kconfig:77]: When do you use `AMEBASMART` or `RTL8730E`?

Comment (kaikaikaime) [reply os/arch/arm/src/amebasmart/Kconfig]: I modified the condition now to AMEBASMART to match with other peripherals

File: os/arch/arm/src/amebasmart/amebasmart_spi.c [hunk 2]
```diff
@@ -859,7 +857,7 @@ static uint16_t amebasmart_spi_send(FAR struct spi_dev_s *dev, uint16_t wd)
 	DEBUGASSERT(priv);
 
 	if (priv->role == AMEBASMART_SPI_MASTER) {
-
+
```
Comment (sunghan-chang) [inline os/arch/arm/src/amebasmart/amebasmart_spi.c:860]: Let's revert this line.

File: os/board/rtl8730e/src/rtl8730e_boot.c [hunk 3]
```diff
@@ -131,6 +131,18 @@ void board_i2s_initialize(void)
 void board_spi_initialize(void)
 {
 #ifdef CONFIG_SPI
+
+#if defined(AMEBASMART_SPI0) && (AMEBASMART_SPI0 == 0)
```
Comment (sunghan-chang) [inline os/board/rtl8730e/src/rtl8730e_boot.c:135]: 1. Maybe it's `CONFIG_.....`.
2. If it's `CONFIG_...`, no need `&& (.... == 0)`.

File: os/board/rtl8730e/src/rtl8730e_boot.c [hunk 4]
```diff
@@ -131,6 +131,18 @@ void board_i2s_initialize(void)
...
+	struct spi_dev_s *spi;
+	spi = up_spiinitialize(0);
+
+#ifdef CONFIG_SPI_USERIO
+	if (spi_uioregister(0, spi) < 0) {
+		lldbg("Failed to register SPI0\n");
+	}
+#endif
+#endif
+#if defined(AMEBASMART_SPI1) && (AMEBASMART_SPI1 == 1)
```
Comment (sunghan-chang) [inline os/board/rtl8730e/src/rtl8730e_boot.c:145]: same as previous

File: os/board/rtl8730e/src/rtl8730e_boot.c [hunk 5]
```diff
@@ -131,6 +131,18 @@ void board_i2s_initialize(void)
 void board_spi_initialize(void)
 {
 #ifdef CONFIG_SPI
+
+#if defined(AMEBASMART_SPI0) && (AMEBASMART_SPI0 == 0)
+	struct spi_dev_s *spi;
```
Comment (sunghan-chang) [inline os/board/rtl8730e/src/rtl8730e_boot.c:136]: Your commit allows to enable both spi0 and spi1. Then, this local variable will be defined twice by spi0 and spi1.

File: os/board/rtl8730e/src/rtl8730e_boot.c (hunk 4 above)
Comment (kaikaikaime) [reply os/board/rtl8730e/src/rtl8730e_boot.c]: Modified*

File: os/board/rtl8730e/src/rtl8730e_boot.c (hunk 5 above)
Comment (kaikaikaime) [reply os/board/rtl8730e/src/rtl8730e_boot.c:134]: I have placed the struct at the CONFIG_SPI so it does not redefined

File: os/board/rtl8730e/src/rtl8730e_boot.c [hunk 6]
```diff
@@ -132,13 +132,23 @@ void board_spi_initialize(void)
 {
 #ifdef CONFIG_SPI
 	struct spi_dev_s *spi;
+#ifdef AMEBASMART_SPI0
```
Comment (sunghan-chang) [inline os/board/rtl8730e/src/rtl8730e_boot.c:135]: It should be `CONFIG_....`

Comment (kaikaikaime) [reply os/board/rtl8730e/src/rtl8730e_boot.c]: Added

**Review by sunghan-chang — APPROVED** [review]

**Review by sunghan-chang — APPROVED** [review]

**Comment by sunghan-chang** [comment]
@kaikaikaime We can use SPI with current defconfig because there is no specific SPI configs, but this commit adds them.
As far as I know, SPI1 is used now. Could you make a commit to enable it by default?

**Comment by kaikaikaime** [comment]
@sunghan-chang ,

I will make another commit to modify the deconfig to set default SPI1 to y

**Pull request merged**